lib/gpu/lib/src/buffer.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+import 'dart:nativewrappers';
+import 'dart:typed_data';
+
+/// A reference to a byte range within a GPU-resident [Buffer].
+class BufferView {
+  /// The buffer of this view.
+  final HostBuffer buffer;
+
+  /// The start of the view, in bytes starting from the beginning of the
+  /// [buffer].
+  final int offsetInBytes;
+
+  /// The length of the view.
+  final int lengthInBytes;
+
+  /// Create a new view into a buffer on the GPU.
+  const BufferView(this.buffer,
+      {required this.offsetInBytes, required this.lengthInBytes});
+}
+
+/// [HostBuffer] is a [Buffer] which is allocated on the host (native CPU
+/// resident memory) and lazily uploaded to the GPU. A [HostBuffer] can be
+/// safely mutated or extended at any time on the host, and will be
+/// automatically re-uploaded to the GPU the next time a GPU operation needs to
+/// access it.
+///
+/// This is useful for efficiently chunking sparse data uploads, especially
+/// ephemeral uniform data that needs to change from frame to frame.
+///
+/// Note: Different platforms have different data alignment requirements for
+///       accessing device buffer data. The [HostBuffer] takes these
+///       requirements into account and automatically inserts padding between
+///       emplaced data if necessary.
+class HostBuffer extends NativeFieldWrapperClass1 {
```

Comment:
Oh, do you mean mark native classes as `base` to avoid the problems mentioned in: https://github.com/flutter/flutter/issues/123756? I think the weird class hierarchy solution that `ui.Picture` moved to is something we could safely migrate to if we wind up having the same testing needs.
Marked `HostBuffer` as `base`.